src/operation.ts

```diff
@@ -0,0 +1,41 @@
+// operation.ts
+
+/*
+ * Copyright (c) 2021-2023 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+export interface CreateOperation<Item> {
+  type: 'create';
+  createdOn: string;
+  item: Item;
+}
+
+export interface DeleteOperation {
+  type: 'delete';
+  createdOn: string;
+  name: string;
+  previousCreatedOn: string;
+}
+
+export type Operation<Item> = CreateOperation<Item> | DeleteOperation;
+
+export function getItemsFromOperations<Item extends { name: string }>(
+  operations: Operation<Item>[],
+  at: string,
```

Comment:
it's not clear to me where we enforce the requirement that the date be > 2018-01-01? is there a test for it?